### PR TITLE
[FIX] website, hr_*: fix some squished buttons

### PR DESCRIPTION
**Pull request by chgo-odoo**
=== ISSUE ===

With the new Milk design, we introduced `.border-radius` on buttons. Most of the time the render looks great, but in some cases, it would break the layout.

This is due to the fact that old design sometimes used two buttons next to each other.

This worked fine because they did not have any radius but now it looks like they are compressed and squished.

Website's buttons were a bit specific so biggest changes happen here.

**Before :**
image

**After :**
image

=== AFTER ===

This commit tracks some of these cases to improve the global render of these buttons.

- Requires : https://github.com/odoo/enterprise/pull/41260

task-3329612
part of task-332626

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by chgo-odoo** [comment]
Hi @stefanorigano 👋

Here is a PR about some squished buttons, could you have a look at it? Especially the website one.

Thanks !

**Comment by chgo-odoo** [comment]
@stefanorigano

File: addons/website/static/src/components/dialog/edit_menu.xml [hunk 1]
```diff
@@ -24,20 +24,20 @@
...
                 <span class="js_menu_label o_text_overflow flex-grow-1">
                     <t t-esc="props.menu.fields['name']"/>
                 </span>
                 <span t-if="props.menu.fields['is_mega_menu']" class="badge text-bg-primary">Mega Menu</span>
                 <i t-if="props.menu.is_homepage" class="fa fa-home ms-3" role="img" aria-label="Home" title="Home"/>
             </span>
             <span class="input-group-append">
-                <button t-on-click="edit" type="button" class="btn btn-primary js_edit_menu fa fa-pencil-square-o" aria-label="Edit Menu Item" title="Edit Menu Item"/>
-                <button t-on-click="delete" type="button" class="btn btn-danger js_delete_menu fa fa-trash-o" aria-label="Delete Menu Item" title="Delete Menu Item"/>
+                <button t-on-click="edit" type="button" class="btn btn-primary js_edit_menu fa fa-pencil-square-o rounded-0" aria-label="Edit Menu Item" title="Edit Menu Item"/>
```
Comment (stefanorigano) [inline addons/website/static/src/components/dialog/edit_menu.xml:39]: `input-group-append` is [deprecated since BS5](https://getbootstrap.com/docs/5.1/migration/#forms) ;)

Unwrapping the buttons should be enough to handle border-radius without the need of using classes, could you check ? (note. be aware of JS tests potentially failing)

File: addons/web/static/src/search/control_panel/control_panel.xml [hunk 2]
```diff
@@ -36,7 +36,7 @@
                     <span class="d-none d-xl-block me-auto"/> <!-- Spacer -->
                 </div>
 
-                <div class="o_control_panel_actions d-empty-none d-flex align-items-center justify-content-start order-2 order-lg-1 w-100 w-lg-auto">
+                <div class="o_control_panel_actions d-empty-none d-flex align-items-center justify-content-start gap-1 order-2 order-lg-1 w-100 w-lg-auto">
```
Comment (stefanorigano) [inline addons/web/static/src/search/control_panel/control_panel.xml:39]: It's unclear to me why this change is needed, could you tell me more about this change?
Screenshots maybe?

File: addons/hr_presence/views/hr_employee_views.xml [hunk 3]
```diff
@@ -40,12 +40,12 @@
                 <div class="oe_kanban_content">
                     <div class="oe_kanban_row">
                         <field name="hr_presence_state" invisible="1"/>
-                        <button name="action_set_present" type="object" class="btn btn-primary btn-sm mt8 fa fa-check" title="Set as present" attrs="{'invisible': [('hr_presence_state', '=', 'present')]}"> </button>
-                        <button name="action_set_absent" type="object" class="btn btn-primary btn-sm mt8 fa fa-times" title="Set as absent" attrs="{'invisible': [('hr_presence_state', '!=', 'present')]}"> </button>
+                        <button name="action_set_present" type="object" class="btn btn-primary btn-sm mt8 fa fa-check me-1" title="Set as present" attrs="{'invisible': [('hr_presence_state', '=', 'present')]}"> </button>
```
Comment (stefanorigano) [inline addons/hr_presence/views/hr_employee_views.xml:43]: why not using `gap-1` on the parent?

**Review by stefanorigano — CHANGES_REQUESTED** [review]

File: addons/hr_presence/views/hr_employee_views.xml (hunk 3 above)
Comment (chgo-odoo) [reply addons/hr_presence/views/hr_employee_views.xml:43]: This was to avoid changing the display property of the parent since it's using a `d-block` and we can't apply `.gap` on top of that.

File: addons/web/static/src/search/control_panel/control_panel.xml (hunk 2 above)
Comment (chgo-odoo) [reply addons/web/static/src/search/control_panel/control_panel.xml:39]: image
image

This fixes the spacing issue in Consolidation > Consolidated Balance

File: addons/hr_presence/views/hr_employee_views.xml (hunk 3 above)
Comment (stefanorigano) [reply addons/hr_presence/views/hr_employee_views.xml:43]: ok... can we `.d-flex.gap-1`?

File: addons/web/static/src/search/control_panel/control_panel.xml (hunk 2 above)
Comment (chgo-odoo) [reply addons/web/static/src/search/control_panel/control_panel.xml:39]: I'm facing huge issues with my localhost. Nothing compiles automatically. I have to restart a DB everytime I do a change...

Meanwhile in managed to find the file where these buttons are located to I managed to apply the changed only there

**Comment by chgo-odoo** [comment]
@stefanorigano Changes done, could you have a look at it please? Note that I opened an Enterprise PR for the Consolidation part. Link can be found in the PR description, thanks 👍

File: addons/hr_presence/views/hr_employee_views.xml [hunk 4]
```diff
@@ -38,13 +38,13 @@
             </xpath>
             <xpath expr="//div[hasclass('oe_kanban_details')]" position="inside">
                 <div class="oe_kanban_content">
-                    <div class="oe_kanban_row">
+                    <div class="oe_kanban_row d-flex gap-1">
```
Comment (stefanorigano) [inline addons/hr_presence/views/hr_employee_views.xml:41]: Good, thanks 👍 
To improve even further let's remove `mt-8` from the buttons and apply a `mb-[x]` to the `oe_kanban_row` or, even better,  applying a `gap-1` to `oe_kanban_content` instead... I'll let you decide according to the visual result on screen ;)

Since we're here, I propose to update this design to `milk`, meaning convert all the buttons to `secondary` and, eventually, use `success` for "Set as present" and `warning` for "Set as absent"

File: addons/project_todo/static/src/scss/todo.scss [hunk 5]
```diff
@@ -3,7 +3,6 @@
         display: flex;
         flex-direction: column;
         .o_form_statusbar {
-            margin: 0;
```
Comment (stefanorigano) [inline addons/project_todo/static/src/scss/todo.scss:6]: Not sure why this change is needed, the visual result seems the same.

File: addons/website/static/src/components/dialog/edit_menu.xml [hunk 6]
```diff
@@ -24,21 +24,17 @@
...
+                <span class="input-group-text bg-view fa fa-bars" role="img" aria-label="Dropdown menu" title="Dropdown menu"/>
+            <span class="form-control d-flex align-items-center bg-view">
                 <span class="js_menu_label o_text_overflow flex-grow-1">
                     <t t-esc="props.menu.fields['name']"/>
                 </span>
                 <span t-if="props.menu.fields['is_mega_menu']" class="badge text-bg-primary">Mega Menu</span>
                 <i t-if="props.menu.is_homepage" class="fa fa-home ms-3" role="img" aria-label="Home" title="Home"/>
             </span>
-            <span class="input-group-append">
                 <button t-on-click="edit" type="button" class="btn btn-primary js_edit_menu fa fa-pencil-square-o" aria-label="Edit Menu Item" title="Edit Menu Item"/>
```
Comment (stefanorigano) [inline addons/website/static/src/components/dialog/edit_menu.xml:36]: Good, thanks 👍 
Last remark, I think that these buttons should be `secondary` now ;)

**Review by stefanorigano — CHANGES_REQUESTED** [review]

File: addons/project_todo/static/src/scss/todo.scss (hunk 5 above)
Comment (chgo-odoo) [reply addons/project_todo/static/src/scss/todo.scss:6]: This was not supposed to be here, it's related to the statusbar task

**Review by stefanorigano — COMMENTED** [review]
Thanks @chgo-odoo 👍 

I find the "Delete" buttons too emphasised tho
image
 
 @pierre-pde , what do you think about setting them `secondary`as well? 
 
image

**Comment by pierre-pde** [comment]
@chgo-odoo & @stefanorigano 
I think it's a good idea to have 'delete' secondary as well :)

**Review by pparidans — APPROVED** [review]
LGTM
@robodoo r+
